source/guides/recipes/autoformatting.md

```diff
@@ -0,0 +1,55 @@
+(autoformatting)=
+# Autoformatting
+
+[`nixfmt`](https://github.com/NixOS/nixfmt) is the official Nix autoformatter.
+It is not yet wired up to any official tooling. Subscribe to
+[NixOS/nix PR #11252](https://github.com/NixOS/nix/pull/11252) for updates on that effort.
+
+Because `nixfmt` doesn't support formatting whole directory trees, you need
+additional tooling such as `treefmt`. The `nixfmt-tree` package provides a
+`treefmt` pre-configured to run `nixfmt` on all nix files in your project. Just
+add it to your shell:
+
+```nix
+mkShell {
+  packages = [ pkgs.nixfmt-tree ];
+}
+```
+
+Note: this assumes you're working in a git repository with a [`.git/index` file
+at the
+root](https://github.com/NixOS/nixpkgs/blob/bbcdf758d61b6095693d00dea8a75061bf89b264/pkgs/by-name/ni/nixfmt-tree/package.nix#L33-L34).
```

Comment:
:+1: for mentioning, however while it's currently correct, it will soon be outdated.

https://github.com/numtide/treefmt/pull/571 changed treefmt's default behaviour, so now it will dynamically ask `git` for the root of the workspace.

With that change, `tree-root-file` is no longer needed in most cases. Therefore, `nixfmt-tree` will remove its `tree-root-file = ".git/index"` definition once the next treefmt version is released.

It will _still_ be correct to say something like:

> this assumes you're project is in a git repository, and you wish to treat the entire repo as your project to be formatted

, however it will no longer be necessary to mention `.git/index` specifically or the implications that has for git worktrees.